lib/routes/security_pin/remote_server_auth.dart

```diff
@@ -148,33 +157,39 @@ class RemoteServerAuthPageState extends State<RemoteServerAuthPage> {
-                                    .dialogTheme
-                                    .contentTextStyle,
-                              ),
-                              optionText: texts
-                                  .remote_server_onion_warning_dialog_default_action_cancel,
-                              optionFunc: () {
-                                Navigator.of(context).pop();
-                              },
-                              okText: texts
-                                  .remote_server_onion_warning_dialog_settings,
-                              okFunc: () async {
-                                connectionWarningResponse = false;
-                                Navigator.of(context).push(FadeInRoute(
-                                  builder: (_) =>
-                                      withBreezTheme(context, NetworkPage()),
-                                ));
-                                // Navigator.of(context).popUntil((route) => route is RemoteServerAuthPage);
-                                return false;
-                              });
+                        if (uri.host.endsWith('onion')) {
+                          if (widget._torBloc.torConfig == null) {
+                            await promptError(
+                                context,
+                                texts.remote_server_warning_connection_title,
+                                Text(
+                                  texts.remote_server_warning_onion_message,
+                                  style: Theme.of(context)
+                                      .dialogTheme
+                                      .contentTextStyle,
+                                ),
+                                optionText: texts
+                                    .remote_server_onion_warning_dialog_default_action_cancel,
+                                optionFunc: () {
+                                  // TODO ubbabeck don't attempt to query
+                                  // navigator pop something
+                                  return;
+                                },
+                                okText: texts
+                                    .remote_server_onion_warning_dialog_settings,
+                                okFunc: () async {
+                                  connectionWarningResponse = false;
+                                  Navigator.of(context).push(FadeInRoute(
+                                    builder: (_) =>
+                                        withBreezTheme(context, NetworkPage()),
+                                  ));
+                                  // Navigator.of(context).popUntil((route) => route is RemoteServerAuthPage);
+                                  return false;
+                                });
+                          } else if (widget._torBloc.torConfig != null) {
+                            await _breezLib
```

Comment:
Why do we need this line? Don't we require restart anyway?

Reply:
removed since we moved the logic to accountbloc as you recommend.